pkg/kotsadm/objects/scripts/export-minio-data.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -e
+
+# This script exports configuration, bucket metadata, iam settings, and bucket content from the old minio instance
+# to the shared migration directory
+
+# check if the migration has already been completed
+if [ -f /export/.migration ];
+then
+    echo "migration already completed, no-op"
+    exit 0
+fi
+
+# validate environment variables
+if [ -z $KOTSADM_MINIO_MIGRATION_WORK_DIR ] ||
+   [ -z $KOTSADM_MINIO_LEGACY_ALIAS ] ||
+   [ -z $KOTSADM_MINIO_ENDPOINT ] ||
+   [ -z $MINIO_ACCESS_KEY ] ||
+   [ -z $MINIO_SECRET_KEY ] ||
+   [ -z $KOTSADM_MINIO_BUCKET_NAME ];
+then
+    echo 's3 migration environment variables not set'
+    exit 1
+fi
+
+# change the working directory to the migration directory
+cd $KOTSADM_MINIO_MIGRATION_WORK_DIR
```

Comment:
IMO this directory should be cleaned up in case there is left over data from a previously failed migration. Maybe that should happen in the first init container before copying the `mc` binary.